siliconcompiler/core.py

```diff
@@ -1688,43 +1688,35 @@ def check_manifest(self):
+        for (step, index) in flowgraph_nodes:
+            in_job = self._get_in_job(step, index)
 
-                for in_step, in_index in self.get('flowgraph', flow, step, index, 'input'):
-                    if in_job != self.get('option', 'jobname'):
-                        workdir = self._getworkdir(jobname=in_job, step=in_step, index=in_index)
-                        cfg = os.path.join(workdir, 'outputs', f'{design}.pkg.json')
-                        if not os.path.isfile(cfg):
-                            self.logger.error(f'{step}{index} relies on {in_step}{in_index} '
-                                              f'from job {in_job}, but this task has not been run.')
-                            error = True
-                        continue
-                    if in_step in steplist and in_index in indexlist[in_step]:
-                        # we're gonna run this step, OK
-                        continue
-                    if self.get('flowgraph', flow, in_step, in_index, 'status') == \
-                       NodeStatus.SUCCESS:
-                        # this task has already completed successfully, OK
-                        continue
-                    self.logger.error(f'{step}{index} relies on {in_step}{in_index}, '
-                                      'but this task has not been run and is not in the '
-                                      'current steplist.')
-                    error = True
+            for in_step, in_index in self.get('flowgraph', flow, step, index, 'input'):
+                if in_job != self.get('option', 'jobname'):
+                    workdir = self._getworkdir(jobname=in_job, step=in_step, index=in_index)
+                    cfg = os.path.join(workdir, 'outputs', f'{design}.pkg.json')
+                    if not os.path.isfile(cfg):
+                        self.logger.error(f'{step}{index} relies on {in_step}{in_index} '
+                                          f'from job {in_job}, but this task has not been run.')
+                        error = True
+                    continue
+                if in_step in steplist and (not indexlist or in_index in indexlist):
+                    # we're gonna run this step, OK
+                    continue
+                if self.get('flowgraph', flow, in_step, in_index, 'status') == NodeStatus.SUCCESS:
+                    # this task has already completed successfully, OK
+                    continue
+                self.logger.error(f'{step}{index} relies on {in_step}{in_index}, '
+                                  'but this task has not been run and is not in the '
+                                  'current steplist.')
+                error = True
 
         # 2. Check library names
         libraries = set()
         for val, step, index in self.schema._getvals('asic', 'logiclib'):
-            if step in steplist and index in indexlist[step]:
+            if step in steplist and (not indexlist or in_index in indexlist):
```

Comment:
why not check for `(step,index) in nodes` instead of using the index list

Reply:
it seems like the get_flowgraph_nodes will return based on the indexlist, so why not just use that where you have checks for indexlist?

Reply:
Just replaced it there.
At some other points there is some logic around the lines of: `if` at least one index of this step or all indices of this step
There I didn't find a prettier replacement yet.